pipert/core/shared_memory.py

```diff
@@ -0,0 +1,130 @@
+import mmap
+import posix_ipc
+
+
+class MemoryIdGenerator:
+    """
+    Generates a new id for a unique shared memory each time get_next is
+    called. The id's are in the following format:
+    "{component_name}_{serial_memory_number}", this helps make a unique
+    shared memory each time even with threads so that we won't override
+    any existing shared memories that were already created.
+    """
+    def __init__(self, component_name, max_count):
+        self.component_name = component_name
+        self.name_count = 0
+        self.max_count = max_count
+
+    def get_next(self):
+        """
+        Generates the next id to use for the shared memory and return
+        the name of a shared memory to free if necessary.
+        """
+        next_name = "{0}_{1}".format(self.component_name, self.name_count)
+        name_to_unlink = ""
+        self.name_count += 1
+        if self.name_count >= self.max_count:
+            name_to_unlink = "{0}_{1}".format(self.component_name,
+                                              (self.name_count
+                                               - self.max_count))
+
+        return next_name, name_to_unlink
```

Comment:
@Elon-Abulafia could you help me understand what is the difference between `next_name` and `name_to_unlink`? 
thanks

Reply:
@ItamarWilf every routine has a max size of shared memory objects in the same time.
if the object name_to_unlink is still alive when we want to allocate a new shared memory object next_name, we release the name_to_unlink object and replace it with next_name